src/server/db_slice.cc

```diff
@@ -745,6 +750,23 @@ size_t DbSlice::DbSize(DbIndex db_ind) const {
   return 0;
 }
 
+string_view DbSlice::GetLockKey(string_view key) {
+  thread_local bool is_enabled_flag_cache = []() {
+    bool value = absl::GetFlag(FLAGS_lock_on_hashtags);
+    if (value && !ClusterConfig::IsEnabledOrEmulated()) {
+      LOG(ERROR) << "Setting --lock_on_hashtags without --cluster_mode is unsupported";
+      quick_exit(1);
```

Comment:
what's `quick_exit` ?

Reply:
It's a standard C++ function ([see here](https://en.cppreference.com/w/cpp/utility/program/quick_exit)) that does not run the regular exit handlers.
One _could_ register special `at_quick_exit` callbacks (similar to `atexit`), but I don't think we do that.
The reason I call it and not `exit()` is to not print the callstack due to a Fiber `CHECK()` that we do not exit from the main thread (I think you said you might fix that?)

Reply:
I prefer not to have the workarounds around the buggy helio. I will try to reproduce it there.

Reply:
Are you trying to get the if to match? If so, run `./dragonfly --logtostdout --lock_on_hashtags`
(tls is unrelated in this case)
Change `quick_exit()` to `exit()` to get the Fiber-related `CHECK()` to fail

Reply:
ok, I succeeded to reproduce it  - thanks! Now I have several comments about the approach.
1. I am fine with flags checking + exit during startup. I think startup time has greater control coverage and if a process stops it's either shown on the console or could be easily tracked using automatic alerts. I think running "exit" in the middle of the flow is not a good methodology and I avoid it at all cost (unless I have to crash the process due to inrecoverable bug). In short - we do flag validation on startup and never exit afterwards. This is not what happens here.
2. I liked your thread_local trick with initialization. Having said that, think it's better to pull this logic out of dbslice for multitude reasons. (1) is one. Another one - performance: for cluster mode you run `ClusterConfig::KeyTag` every time you access lock keys again and again. Lastly, I think db_slice  does not have to know about cluster mode and for that, we have KeyLockArgs that may contain `args` already pointing to extracted slots. Also see (3) for another reason to do it at a higher level.
3. Consider a case `mset {s}a val {s}b val` - how many times you want to lock `{s}` and how it will affect the transaction logic? specifically mset will never be marked as OOO. we even have a unit test somewhere that tests "mset a val a val2" for the same reason. 

I think this PR deserves a dedicated unit test too.

Reply:
As discussed offline:
re/ (1): I moved the check to startup, much better
re/ (2): I moved the normalization to KeyLockArgs, but have _not_ changed it to hold normalized keys as that would be a big refactor
re/ (3): that's irrelevant as locks are checked via maps within transaction
I also added a UT which will sets and gets hashtag keys, and it will run (like all other UTs) in all 3 modes: no cluster, emulated cluster and emulated cluster + lock hashtags.